src/app/test/test.component.ts

```diff
@@ -113,37 +115,43 @@ export class TestComponent implements OnInit, AfterViewInit {
 
   getCopiedReports(): void {
     this.httpService.getTestReports(this.currentView.metadataNames, this.currentView.storageName).subscribe({
-      next: (response) => this.addCopiedReports(response),
+      next: (response: TestListItem[]) => this.addCopiedReports(response),
       error: () => catchError(this.httpService.handleError()),
     });
   }
 
-  loadData(path: any): void {
-    this.httpService.getViews().subscribe((views) => {
-      const defaultViewKey = Object.keys(views).find((view) => views[view].defaultView);
+  loadData(path: string): void {
+    this.httpService.getViews().subscribe((views: Record<string, View>) => {
+      const defaultViewKey: string | undefined = Object.keys(views).find((view: string) => views[view].defaultView);
       if (defaultViewKey) {
-        const selectedView = views[defaultViewKey];
-        this.currentView.targetStorage = selectedView.storageName;
+        const selectedView: View = views[defaultViewKey];
+        if (selectedView.storageName) {
+          this.currentView.targetStorage = selectedView.storageName;
+        }
       }
     });
     this.httpService.getTestReports(this.currentView.metadataNames, this.currentView.storageName).subscribe({
-      next: (value) => {
+      next: (value: TestListItem[]) => {
         this.reports = value;
         this.testFileTreeComponent.setData(this.reports);
         if (path) {
           setTimeout(() => {
             this.testFileTreeComponent.selectItem(path);
           });
         } else {
-          setTimeout(() => {
-            this.testFileTreeComponent.tree.selectItem(this.testFileTreeComponent.rootFolder.name);
-          });
+          this.loadRootData();
```

Comment:
Thank you for the change! But I wonder what would happen if `this.testFileTreeComponent.rootFolder.name` is empty. Would this get stuck in an infinite loop? In other cases, if the `path` is empty the method will be called again, having to subscribe (fetch data) again, being a bit wasteful, isn't it.
Maybe best to don't do anything (and trow an error) if a path is missing? (like what are you going to select then anyway :P). Also, `selectPath` might be a bit more descriptive than `path`.

Reply:
I've looked into it and it does look like loadRootData is getting stuck in an infinite loop, but on the `ngOnInit` and the `ngAfterViewInit` already. I'm not sure I understand why this is happening, because when the code from `loadRootData` was still in `loadData` it was fine.

Reply:
Oh loadRootData only contains
```ts
loadRootData(): void {
    setTimeout(() => {
      this.testFileTreeComponent.tree.selectItem(this.testFileTreeComponent.rootFolder.name);
    });
  }
```
I thought it was
```ts
loadRootData(): void {
    loadData(this.testFileTreeComponent.rootFolder.name)
}
```
That would get stuck in a loop. But it is not like that, so never mind.

Reply:
`this.testFileTreeComponent.rootFolder.name` is also selected differently than when you give `loadData` a `selectPath` input. 
For `loadData` the `selectItem` function is being called on `this.testFileTreeComponent` while for `this.testFileTreeComponent.rootFolder.name` the `selectItem` function is called on ``this.testFileTreeComponent.tree`.

Also with the current iteration of `loadRootData()` it is still getting stuck in an infinite loop. Is there anything I can do to fix this or should i put it back into the `loadData` function?